### PR TITLE
Update example code in IDBObjectStore.delete() doc

### LINKED ISSUE #5084: Issue with "IDBObjectStore.delete()": In the example sample, it uses a global db variable.
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/delete

#### What information was incorrect, unhelpful, or incomplete?
In the example  code sample, it uses a global db variable
```db = DBOpenRequest.result;```

#### Specific section or headline?

https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/delete#example
in the onsucces function ```DBOpenRequest.onsuccess = function(event) {```

#### What did you expect to see?
This should either become
```window.db =```

Or at the top under the ```var DBOpenRequest``` declaration, a ```var db``` line.

#### Did you test this? If so, how?

No, I just saw it.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/idbobjectstore/delete`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/delete
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/idbobjectstore/delete/index.html
* Last commit: https://github.com/mdn/content/commit/12024ee393c3fe3101b03903561c640444b38646
* Document last modified: 2021-04-08T14:57:07.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/5084

### CODE REVIEW

File: files/en-us/web/api/idbobjectstore/delete/index.html [hunk 1]
```diff
@@ -91,8 +91,11 @@ <h2 id="Example">Example</h2>
     href="https://github.com/mdn/to-do-notifications/">To-do Notifications</a> app
     (<a href="https://mdn.github.io/to-do-notifications/">view example live</a>.)</p>
 
-<pre class="brush: js">// Let us open our database
-var DBOpenRequest = window.indexedDB.open("toDoList", 4);
+<pre class="brush: js">
+// create an instance of a db object to store the IDB data in
+const db = DBOpenRequest.result;
```
Comment (wbamberg) [inline files/en-us/web/api/idbobjectstore/delete/index.html:96]: Isn't `DBOpenRequest` going to be undefined here?

**Review by wbamberg — CHANGES_REQUESTED** [review]
This is a tricky one because the example is part of a bigger example which is at https://github.com/mdn/to-do-notifications/blob/gh-pages/scripts/todo.js (and which has already got out of sync with the snippet in this page, one of the perils of using [`EmbedGHLiveSample`](https://github.com/mdn/yari/blob/main/kumascript/macros/EmbedGHLiveSample.ejs)).

In the bigger example `db` is declared global: https://github.com/mdn/to-do-notifications/blob/gh-pages/scripts/todo.js#L6 , but even with that addressed, this example won't be runnable on its own - for example it also uses `note` which will be undefined in this snippet.

Perhaps if we changed the `note` bits to use `console.log` it might be better as a standalone example? Or maybe it would be better to lean into the GH example, by copying https://github.com/mdn/to-do-notifications/blob/gh-pages/scripts/todo.js#L218-L232 instead, and in the text around this example explained where `db` and `note` are supposed to be coming from, and how this snippet fits into the overall application at https://github.com/mdn/to-do-notifications/blob/gh-pages/scripts/todo.js ?

File: files/en-us/web/api/idbobjectstore/delete/index.html (hunk 1 above)
Comment (sideshowbarker) [inline files/en-us/web/api/idbobjectstore/delete/index.html:96]:
```suggestion
const db;
```

Comment (sideshowbarker) [reply files/en-us/web/api/idbobjectstore/delete/index.html]: > Isn't `DBOpenRequest` going to be undefined here?

oops — yeah, fixed now

File: files/en-us/web/api/idbobjectstore/delete/index.html [hunk 2]
```diff
@@ -91,8 +91,11 @@ <h2 id="Example">Example</h2>
     href="https://github.com/mdn/to-do-notifications/">To-do Notifications</a> app
     (<a href="https://mdn.github.io/to-do-notifications/">view example live</a>.)</p>
 
-<pre class="brush: js">// Let us open our database
-var DBOpenRequest = window.indexedDB.open("toDoList", 4);
+<pre class="brush: js">
+// create an instance of a db object to store the IDB data in
+const db;
```
Comment (sideshowbarker) [inline files/en-us/web/api/idbobjectstore/delete/index.html:96]:
```suggestion
let db;
```

**Comment by sideshowbarker** [comment]
> Perhaps if we changed the `note` bits to use `console.log` it might be better as a standalone example? Or maybe it would be better to lean into the GH example, by copying [mdn/to-do-notifications@`gh-pages`/scripts/todo.js#L218-L232](https://github.com/mdn/to-do-notifications/blob/gh-pages/scripts/todo.js#L218-L232) instead, and in the text around this example explained where `db` and `note` are supposed to be coming from, and how this snippet fits into the overall application at [mdn/to-do-notifications@`gh-pages`/scripts/todo.js](https://github.com/mdn/to-do-notifications/blob/gh-pages/scripts/todo.js) ?

Those all sound like great improvements to me, but the only thing I was attempting to do in this PR was just to resolve the specific issue the OP of #5084 pointed out. The other improvements will need more time than I’m going to personally be able to put in on this one.

So we could either merge this PR and raise a new issue for follow-up changes, or we can close this PR unmerged and add a comment to #5084 noting the additional changes needed, and block that issue on those changes getting made — and I can un-assign that issue and label it with “help wanted” to encourage someone else to pick it up to get it resolved.

**Comment by wbamberg** [comment]
That's fair enough. Since I'm the one complaining I tried a slightly different approach in https://github.com/mdn/content/pull/6013, that just uses the code from the example app directly and tries to explain its context a bit.

If you like it, please merge it, or if you like this more minimal change better we could just merge this PR as it is.

**Comment by sideshowbarker** [comment]
> I tried a slightly different approach in #6013, that just uses the code from the example app directly and tries to explain its context a bit.

Wow yeah that’s how it should be

> If you like it,

I like it :) Merged — much thanks

**Comment by sideshowbarker** [comment]
Closing as superseded by https://github.com/mdn/content/pull/6013

**Pull request closed without merging**